docs/guides/enforce-active-organization.mdx

```diff
@@ -0,0 +1,208 @@
+
+  ```typescript {{ filename: '/middleware.ts', mark: [[7, 13]] }}
+  import { clerkMiddleware, createRouteMatcher } from '@clerk/nextjs/server'
+  import { NextRequest, NextResponse } from 'next/server'
+
+  const isPublicRoute = createRouteMatcher(['/', '/sign-in', '/sign-up'])
+
+  export default clerkMiddleware((auth, req: NextRequest) => {
+    // Check if the user is signed in and does not have an auth().orgId
+    // If true, then redirect them to the organization selection route
+    // Include their current route as a redirect_url, which will be handled on the route
+    if (auth().userId && !auth().orgId && req.nextUrl.pathname !== '/organization-selection') {
+      const orgListUrl = new URL('/organization-selection', req.url)
+      return NextResponse.redirect(orgListUrl)
+    }
+
+    // If the user isn't signed in and the route is private, redirect to sign-in
+    if (!auth().userId && !isPublicRoute(req))
+      return auth().redirectToSignIn({ returnBackUrl: req.url })
+
+    // If the user is logged in and the route is protected, let them view.
+    if (auth().userId && !isPublicRoute(req)) return NextResponse.next()
+  })
+
+  export const config = {
+    matcher: ['/((?!.*\\..*|_next).*)', '/', '/(api|trpc)(.*)'],
+  }
+  ```
+
+  ### Setup the `/organization-selection` route
+
+  The `/ogranization-selection` route will render the `<OrganizationList />` component. The `hidePeronsal` prop is passed, which will hide the option for a user to select their Personal Worksapce. This will mean that all users, even new users, will need to either create or join and organization.
+
+  The component will also check the `orgId` from `auth()`. This will be falsy when the user is first redirected to the route. When the user creates or joins an organization, the auth object will be refreshed and `orgId` will have a valid organization id. When that happens the user will be redirected to `/dashboard`.
+
+  ```tsx {{ filename: '/app/organization-selection/page.tsx' }}
+  import { OrganizationList } from '@clerk/nextjs'
+  import { auth } from '@clerk/nextjs/server'
+  import { redirect } from 'next/navigation'
+
+  export default async function OrganizationSelectionPage() {
+    const { orgId } = auth()
+
+    // Once the `orgId` is set, redirect the user. In example the user will be redirected to `/dashboard`
+    if (orgId) {
+      redirect('/dashboard')
+    }
+
+    return (
+      <>
+        <h1>Select an organization</h1>
+        <OrganizationList hidePersonal />
+      </>
+    )
+  }
+  ```
+
+  ### Improving the user experience with redirects
+
+  When a user does not have an active organization and is redirected, we can add a `redirect_url` to the URL and then later handle the direct on the `/organization-selection` route.
+
+  ### Updating Middleware
+
+  Let's start with the changes to Middleware. The Middleware will now read the `req.nextUrl.href` and URL endcode it, and then add that as a `redirect_url` search parameter.
```

Comment:
```suggestion
  #### Update the Middleware

  Modify `clerkMiddleware()` to include the `redirect_url` search parameter in the URL when redirecting to the `/organization-selection` route. This will allow that route to consume the search parameter and use it to redirect the user back to the route they were trying to access once they have created or joined an organization.
```